kafka_helpers/consumer.go

```diff
@@ -0,0 +1,50 @@
+package kafka_helpers
+
+import (
+	"context"
+	"fmt"
+	"github.com/segmentio/kafka-go"
+	"internship_project/elasticsearch_helpers"
+	"log"
+	"strings"
+)
+
+type KafkaConsumer struct {
+	Reader   *kafka.Reader
+	EsClient elasticsearch_helpers.ElasticsearchClient
+}
+
+func (consumer *KafkaConsumer) Consume() {
+	fmt.Println("KafkaConsumer is ready to consume new messages.")
+	for {
+
+		m, err := consumer.Reader.FetchMessage(context.Background())
+		if err != nil {
+			log.Printf("Error while fetching message")
+		}
+
+		fmt.Printf("message at offset %d: %s = %s\n", m.Offset, string(m.Key), string(m.Value))
+
+		s := strings.Split(string(m.Value), " ")
+
+		if len(s) == 3 {
+			if s[0] == "CREATED" || s[0] == "UPDATED" {
+				go consumer.EsClient.IndexDocument(string(m.Key), s[2])
+			} else if s[0] == "DELETED" {
+				go consumer.EsClient.DeleteDocument(string(m.Key))
+			}
+		} else {
+			log.Printf("Failed to parse Kafka message")
+		}
+
+		err = consumer.Reader.CommitMessages(context.Background(), m)
+		if err != nil {
```

Comment:
If a Kafka message fails to commit, you will fetch it again in the next for loop. This means that you can potentially have a large number of duplicates, due to the fact that the operation carried by the message is executed again each time. Any ideas how to handle this?